source/line.h

```diff
@@ -1,10 +1,12 @@
 #pragma once
 #include "common.h"
+#include "point.h"
+#include "constants_geom.h"
 
 struct Line{
 	ld a, b, c; // b = 1 non-vertical lines, b = 0 vertical lines
 	Line(ld a_, ld b_, ld c_) : a(a_), b(b_), c(c_) {}
-	Line(Point<ll>& p1, Point<ll>& p2){
+	Line(const Point<ld>& p1, const Point<ld>& p2){
```

Comment:
I'm not too sure about changing this to `ld` by default, as some problems might require `ll`. Would making `Line` templated work? Something like

```
template<typename T>
struct Line {
    Line(const Point<T>& p1, ...) {}
    [...]
```

Reply:
Why do you say having the template before the class (as we do for `Point`) doesn't work? Btw, if we don't want everything as a template, we could use something like `using P = Point<ld>;` before the class.

Reply:
if line is a template most occurrences of `Line` in the code have to become `Line<T>`. I think it's better like this with templates on single functions. After all the class structure is always the same except for when you need to use Point<T>. 
We can do something like `#define TTT template<typename T>` and that saves a lot of lines

Reply:
Still not a huge fan of templating everything. But this is a minor problem imo, so there's no need to block the PR.

Reply:
I added another comment on this below, but fundamentally we probably don't need templating here at all

Reply:
ok, back to ld then. I think that's for the best. Doesn't bug in weird ways, we still always have ld precision